sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/GcpApiSurfaceTest.java

```diff
@@ -0,0 +1,84 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.gcp;
+
+import static org.apache.beam.sdk.util.ApiSurface.Matchers.classesInPackage;
+import static org.apache.beam.sdk.util.ApiSurface.Matchers.containsOnlyClassesMatching;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import com.google.cloud.bigtable.grpc.BigtableInstanceName;
+import com.google.cloud.bigtable.grpc.BigtableTableName;
+import com.google.common.collect.ImmutableSet;
+import java.io.IOException;
+import java.util.Set;
+import org.apache.beam.sdk.util.ApiSurface;
+import org.hamcrest.Matcher;
+import org.hamcrest.Matchers;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/**
+ * API surface verification for {@link org.apache.beam.sdk.io.gcp}.
+ */
+@RunWith(JUnit4.class)
+public class GcpApiSurfaceTest {
+
+  @SuppressWarnings("unchecked")
+  private final Set<Matcher<Class<?>>> allowedClasses =
+      ImmutableSet.of(classesInPackage("com.google.api.client.json"),
+                      classesInPackage("com.google.api.client.util"),
+                      classesInPackage("com.google.api.services.bigquery.model"),
+                      classesInPackage("com.google.auth"),
+                      classesInPackage("com.google.bigtable.v2"),
+                      classesInPackage("com.google.cloud.bigtable.config"),
+                      Matchers.<Class<?>>equalTo(BigtableInstanceName.class),
+                      Matchers.<Class<?>>equalTo(BigtableTableName.class),
+                      // https://github.com/GoogleCloudPlatform/cloud-bigtable-client/pull/1056
+                      classesInPackage("com.google.common.collect"),
+                      // via Bigtable, PR above out to fix.
+                      classesInPackage("com.google.datastore.v1"),
+                      classesInPackage("com.google.protobuf"),
+                      classesInPackage("com.google.type"),
+                      classesInPackage("com.fasterxml.jackson.annotation"),
+                      classesInPackage("com.fasterxml.jackson.core"),
+                      classesInPackage("com.fasterxml.jackson.databind"),
+                      classesInPackage("io.grpc"),
+                      classesInPackage("java"),
+                      classesInPackage("javax"),
+                      classesInPackage("org.apache.beam"),
+                      classesInPackage("org.apache.commons.logging"),
+                      // via Bigtable
+                      classesInPackage("org.joda.time"));
+
+  private final ApiSurface apiSurface =
+      ApiSurface
+          .ofPackage(getClass().getPackage().getName())
+          .pruningPattern("org[.]apache[.]beam[.].*Test.*")
+          .pruningPattern("org[.]apache[.]beam[.].*IT")
+          .pruningPattern("java[.]lang.*");
+
+  // required on account of the apiSurface field
```

Comment:
Maybe expand this slightly. What is the difference between this and `SdkCoreApiSurfaceTest`, which does not have it. You can move exception causing code into `testGcpApiSurface` since it isn't reused, or also convert to a static initialization block, or move the field assignment into the constructor. It might be slightly confusing that the exception declared here has to do with things that happen elsewhere. If it is thoroughly explained, then it is OK.